docs/content/concepts/assets/software-defined-assets.mdx

```diff
@@ -63,7 +71,91 @@ def downstream_asset(upstream_asset):
     return upstream_asset + [4]
 ```
 
-The [explicit dependencies](#explicit-dependencies) example covers an alternative way to specify asset dependencies without needing to match argument names to upstream asset names.
+#### Defining explicit dependencies
+
+If defining dependencies by matching argument names to upstream asset names feels too magical for your tastes, you can also define dependencies in a more explicit way:
+
+```python file=/concepts/assets/explicit_asset_dependency.py
+from dagster import AssetIn, asset
+
+
+@asset
+def upstream_asset():
+    return [1, 2, 3]
+
+
+@asset(ins={"upstream": AssetIn("upstream_asset")})
+def downstream_asset(upstream):
+    return upstream + [4]
+```
+
+In this case, `ins={"upstream": AssetIn("upstream_asset")}` declares that the contents of the asset with the key `upstream_asset` will be provided to the function argument named `upstream`.
```

Comment:
Maybe it would be good to mention that you can also provide an asset key to `AssetIn` so users can choose to explicitly identify the asset instead of relying on the string inference

Reply:
Both are valid. You could do `AssetIn(asset_key="upstream_asset")` or `AssetIn(asset_key=AssetKey("upstream_asset"))`